### PR TITLE
fix: connect menu

### PR DESCRIPTION
**Pull request by yangej**
### Description
fix broken navbar styles on mobile, and slightly... refactor the code

### Implementation
- 8085cab41da061c1ff295f751ade71ec66b4e2e9 add custom button supporting all navbar styles
- 623d457bdc624c96021538caa99b54a1d0030055 refactor and apply custom button to connect button
- 07d0833218abc5f6edcb8ef9c8ba9679f3ef1419 apply custom button to shop button
- 291636bcc49c9c9a1958afecfff4d9059a39aa66 apply custom button to lang button
- 89ded0c6d22c993e4e2751b1ba301be875f50cd3 fix navbar styles

### Future work
- [ ] add custom button with popper
- [ ] refactor, refactor and refactor lol...

### Relavant links
[https://trello.com/c/tPnhNT5D/46-web-fix-mobile-header-ui](ticket)

### CODE REVIEW

File: src/components/Header/ConnectMenu/ConnectButton.tsx [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+}
+
+interface ConnectButtonProps {
+  status: ConnectButtonStatus;
+}
+const ConnectButton = ({ status }: ConnectButtonProps) => {
+  const isMobile = useMediaQuery(mobileMediaQuery);
+  const { connect, disconnect } = useWeb3Context();
+  if (status === ConnectButtonStatus.Connected) {
+    if (isMobile) return <CustomButton type="outline" text="Disconnect" onClick={disconnect} />;
```
Comment (otterclamking) [inline src/components/Header/ConnectMenu/ConnectButton.tsx]: the text need to apply i18n

File: src/components/Header/ConnectMenu/ConnectButton.tsx [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    return <CustomButton type="outline" text="Disconnect" icon={MetamaskIcon} onClick={disconnect} />;
+  }
+  if (status === ConnectButtonStatus.NotConnected) {
+    if (isMobile) return <CustomButton text="Connect" onClick={connect} />;
```
Comment (otterclamking) [inline src/components/Header/ConnectMenu/ConnectButton.tsx:31]: i18n

File: src/components/Header/ConnectMenu/ConnectButton.tsx [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+  }
+
+  const [popperOpen, setPopperOpen] = useState(false);
+  const buttonRef = useRef(null);
+  if (status === ConnectButtonStatus.InProgress) {
+    return (
+      <>
+        <CustomButton
+          ref={buttonRef}
+          text="In Progress"
```
Comment (otterclamking) [inline src/components/Header/ConnectMenu/ConnectButton.tsx]: i18n

Comment (yangej) [reply src/components/Header/ConnectMenu/ConnectButton.tsx]: I found that 'in progress' status not existing in UI lol..

Comment (yangej) [reply src/components/Header/ConnectMenu/ConnectButton.tsx]: fixed 5a0c6c1

File: src/components/Header/ConnectMenu/ConnectButton.tsx (hunk 2 above)
Comment (yangej) [reply src/components/Header/ConnectMenu/ConnectButton.tsx:31]: fixed 5a0c6c1

File: src/components/Header/ConnectMenu/ConnectButton.tsx (hunk 1 above)
Comment (yangej) [reply src/components/Header/ConnectMenu/ConnectButton.tsx]: fixed 5a0c6c1

descs haven't been translated will be handle in other ticket. Here I just add todo comments for future modification.

**Review by otterclamking — APPROVED** [review]

**Pull request merged**